server/consumer.go

```diff
@@ -1991,8 +1995,12 @@ func (o *consumer) processAck(subject, reply string, hdr int, rmsg []byte) {
 		o.processNak(sseq, dseq, dc, msg)
 	case bytes.Equal(msg, AckProgress):
 		o.progressUpdate(sseq)
-	case bytes.Equal(msg, AckTerm):
-		o.processTerm(sseq, dseq, dc)
+	case bytes.HasPrefix(msg, AckTerm):
+		var reason string
+		if buf := msg[len(AckTerm):]; len(buf) > 0 {
```

Comment:
@scottf had a good question here, do you think we should limit the length of the reason @derekcollison ?

Reply:
Also, the content? Like should we restrict it to printable, should we encode it? Maybe we should wrap in in json {"reason":"blah whatever b/c it's json encoded"}

Reply:
Will be limited by max msg payload. 
Right now see as a closure that R.I. thought would be important.

Reply:
I am not attached to this format. I just made it the same as the others so if someone has a suggestion for change we can do that